File: backend/app/models/ai_assistants.py
from __future__ import annotations

from datetime import datetime

from sqlalchemy import Boolean, CheckConstraint, DateTime, Index, Integer, String, Text, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column

from app.db import Base


class AIAssistantConfig(Base):
    """AI 助手配置：通用助手、部门助手、管理助手等可复用画像."""

    __tablename__ = "ai_assistant_configs"
    __table_args__ = (
        CheckConstraint("scope IN ('global','department')", name="ck_ai_assistant_scope"),
        UniqueConstraint("scope", "department", "name", name="uq_ai_assistant_scope_department_name"),
        Index("idx_ai_assistant_scope", "scope"),
        Index("idx_ai_assistant_department", "department"),
    )

    assistant_id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    scope: Mapped[str] = mapped_column(String, nullable=False, default="global")
    department: Mapped[str | None] = mapped_column(String, nullable=True)
    name: Mapped[str] = mapped_column(String, nullable=False)
    role: Mapped[str] = mapped_column(String, nullable=False, default="management")
    prompt: Mapped[str] = mapped_column(Text, nullable=False)
    workflow: Mapped[str | None] = mapped_column(Text, nullable=True)
    cadence: Mapped[str] = mapped_column(String, nullable=False, default="weekly")
    enabled: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)
    created_by: Mapped[str] = mapped_column(String, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

File: frontend/src/api/aiAssistants.ts
import { api } from "./client";

export type AIAssistantScope = "global" | "department";
export type AIAssistantCadence = "daily" | "weekly" | "manual";

export interface AIAssistantConfig {
  assistant_id: number;
  scope: AIAssistantScope;
  department: string | null;
  name: string;
  role: string;
  prompt: string;
  workflow: string | null;
  cadence: AIAssistantCadence;
  enabled: boolean;
  created_by: string;
  created_at: string;
  updated_at: string;
}

export interface AIAssistantPayload {
  scope: AIAssistantScope;
  department?: string | null;
  name: string;
  role: string;
  prompt: string;
  workflow?: string | null;
  cadence: AIAssistantCadence;
  enabled: boolean;
}

export const listAIAssistants = async (params?: {
  scope?: AIAssistantScope;
  department?: string;
  enabled_only?: boolean;
}): Promise<AIAssistantConfig[]> => {
  const { data } = await api.get<AIAssistantConfig[]>("/ai-assistants", { params });
  return data;
};

export const createAIAssistant = async (payload: AIAssistantPayload): Promise<AIAssistantConfig> => {
  const { data } = await api.post<AIAssistantConfig>("/ai-assistants", payload);
  return data;
};

export const updateAIAssistant = async (
  assistantId: number,
  payload: AIAssistantPayload,
): Promise<AIAssistantConfig> => {
  const { data } = await api.patch<AIAssistantConfig>(`/ai-assistants/${assistantId}`, payload);
  return data;
};
